### PR TITLE
Rework instrumenting page

### PR DESCRIPTION
**Pull request by svrnm**
As discussed a few times I am not very happy with the current state of the Instrumenting page. This is a try to make it better. 

https://deploy-preview-2063--opentelemetry.netlify.app/docs/concepts/instrumenting/

### CODE REVIEW

File: content/en/docs/concepts/instrumenting.md [hunk 1]
```diff
@@ -1,33 +1,54 @@
...
+- for libraries without such an integration the OpenTelemetry projects provide
+  language specific
+  [Instrumentation Libraries](/docs/reference/specification/overview/#instrumentation-libraries).
+
+To facilitate the instrumentation of applications even more,
+[Automatic Instrumentation](/docs/reference/specification/glossary/#automatic-instrumentation)
+can be used to collect telemetry from an application without the end-user being
+required to modify the application's source code. If you previously used an APM
+agent to automatically extract telemetry from your application, Automatic
+Instrumentation will give you a similar out of the box experience.
```
Comment (cartermp) [inline content/en/docs/concepts/instrumenting.md:34]: Hmmm, I feel like maybe this should be flipped: start with automatic instrumentation or instrumentation libraries, and then "to facilitate the instrumentation of applications even more," talk about manual instrumentation.

Comment (svrnm) [reply content/en/docs/concepts/instrumenting.md:34]: but isn't automatic instrumentation the most convenient one? And manual instrumentation is maybe more work but gives you that extra details. Or are you thinking about the flow -> instrument your app automatically -> gradually add manual instrumentation?

Comment (cartermp) [reply content/en/docs/concepts/instrumenting.md:34]: Yes, the latter, the flow of the page should probably lead with automatric instrumentation (and/or instrumentation libraries), followed by manual instrumentation.

Comment (svrnm) [reply content/en/docs/concepts/instrumenting.md]: fixed, how about now?

Comment (cartermp) [reply content/en/docs/concepts/instrumenting.md]: I like it!

**Review by cartermp — APPROVED** [review]

**Pull request merged**